### ISSUE TITLE
Add support for GetHeapStatistics

### ISSUE THREAD

**Issue by dallen66** [denoland/rusty_v8#381]
Can you add support for GetHeapStatistics. I need to monitor the isolate memory usage. I try to implement it my self but failed.  My c++ and binding is not great.

binding.cc
```c
void v8__Isolate__GetHeapStatistics(v8::Isolate* isolate) {
  v8::HeapStatistics stats;
  return isolate->GetHeapStatistics(&stats);
}
```

isolate.rs
```c
extern "C" {
  fn v8__Isolate__GetHeapStatistics(isolate: *mut Isolate);
}
/// Get statistics about the heap memory usage. 
pub fn get_heap_statistics(&mut self) {
  unsafe { v8__Isolate__GetHeapStatistics(self) }
}
```
I get this when I try to run the function.
```
undefined reference to `v8__Isolate__GetHeapStatistics'
          collect2: error: ld returned 1 exit status
```
I build from source with ``V8_FROM_SOURCE=1 cargo build -vv`` Is there some else I need to do.

**Comment by ry** [comment]
@dallen66 The binding to GetHeapStatistics is welcome. If you start a PR with your patch, we can help get it done.

> undefined reference to `v8__Isolate__GetHeapStatistics'
>          collect2: error: ld returned 1 exit status

I believe this is a bug on our side. You need to run "cargo clean" and then "V8_FROM_SOURCE=1 cargo build -vv".

**Comment by dallen66** [comment]
Thank will try that. I will also work on the unit test.

**Comment by dallen66** [comment]
I made the required patches.
https://github.com/denoland/rusty_v8/pull/383

**Comment by dallen66** [comment]
These functions should be moved to the IsolateHandle struct to make them work across threads.
I'll wait until you check what I did before I make any changes.

**Comment by bnoordhuis** [comment]
Sorry, I forgot add a `Fixes` tag but this was fixed by #440.
